Домашняя работа 5.

In [1]:
from math import sqrt 
from math import pi
from math import factorial
from math import exp
from scipy.stats import norm
import numpy as np

norm.cdf, norm.ppf

Задание 1.

1) Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания с надежностью 0.95,если выборочная средняя M = 80, а объем выборки n = 256.

стандартная ошибка среднего равна:

In [2]:
err = 16/sqrt(256)
err

1.0

Z[a/2] для надежности 0.95 = 1,96

In [3]:
Z1 = 80+1.96*1
Z1

81.96

In [4]:
Z2 = 80-1.96*1
Z2

78.04

Доверительный интервал - [78.04, 81.96]

Задание 2.

В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

In [5]:
arr = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])

In [6]:
avg = arr.mean()
avg

6.590000000000001

Критерий стьдента для данной задачи - 2.262

In [7]:
std = sqrt(arr.var(ddof =1))
std

0.4508017549014448

In [8]:
Z1 = avg + (2.262*std)/sqrt(10)
Z1

6.912461744087575

In [9]:
Z2 = avg - (2.262*std)/sqrt(10)
Z2

6.267538255912426

Доверительный интервал - [6.267, 6.912]

Задание 3

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв.мм.

Z[a] для надежности 0.95 = 1,65

In [10]:
Z = (17.5 - 17)/(sqrt(4)/sqrt(100))
Z

2.5

Z[a] < Z => гипотеза неверна

Задание 4

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек.
Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно,
что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

In [11]:
arr = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

In [12]:
avg = arr.mean()
avg

198.5

доверительная вероятность 0.99, значит а = 0.01. Нам нужна а/2 = 0.05. По таблице находим значение критерия стьюдента для выборки из 10 элементов.

Критерий стьюдента = 3,25 

In [14]:
std = sqrt(arr.var(ddof =1))
std

4.453463071962462

In [16]:
T = (avg - 200)/(std/sqrt(10))
T

-1.0651074037450896

так как T лежит в диапазоне от -3,25 до 3,25, считаем что утверждение продавца(нулевая гипотеза) верно. 